tests/ui/or_fun_call.fixed

```diff
@@ -226,7 +226,7 @@ mod issue8239 {
             acc.push_str(&f);
             acc
         })
-        .unwrap_or_default();
```

Comment:
Isn't converting `unwrap_or` with default / `new` values to `unwrap_or_default` the purpose of the lint? Am I missing something?

Reply:
You're not missing anything. This is fallout from using the heuristic introduced in #11135. In particular, no warning is issued because the receiver is considered to have uncertain type: https://github.com/rust-lang/rust-clippy/blob/e27977b84776b641ec48482f18c1dac7420a5830/tests/ui/or_fun_call.fixed#L225-L228

Hopefully you agree this false negative is a worthwhile tradeoff for reducing false positives as described in that PR. (My impression is Clippy tends to err that way.)

Apologies, as I should have called out this tradeoff explicitly. (I'm sure you noticed, but there are three other instances just above this one.)